spec/serializers/api/admin/order_cycle_serializer_spec.rb

```diff
@@ -0,0 +1,27 @@
+require "spec_helper"
+
+module Api
+  module Admin
+    describe OrderCycleSerializer do
+      let(:order_cycle) { create(:order_cycle) }
+      let(:serializer) { Api::Admin::OrderCycleSerializer.new order_cycle, current_user: order_cycle.coordinator.owner }
+
+      it "serializes an order cycle" do
+        expect(serializer.to_json).to match order_cycle.name
+      end
+
+      it "serializes the order cycle with exchanges" do
+        expect(serializer.exchanges.to_json).to include order_cycle.variants.first.id.to_s
```

Comment:
Searching for a random number (id) in a string can easily lead to false positives. I would probably write it like this:

```rb
exchanges = JSON.parse(serializer.exchanges.to_json)
expect(exchanges).to include include({"variants" => {order_cycle.variants.first.id.to_s => true}})
```

I know it's a mind boggling matcher. The first include matches any element of the array and the second include is a partial match of the hash.

A compromise would be to match part of the JSON to avoid the id `8` being matched in `18`:
```rb
expect(serializer.exchanges.to_json).to include "\"order_cycle.variants.first.id\":true"
```

But that requires knowledge of the JSON format used and formatting would break it. And it's still not testing that the id belongs to a variant.

Reply:
yeah, good point! I used your last suggestion 👍